### PR TITLE
Add Naive Bayes Classifier

### PR DESCRIPTION
**Pull request by lanchiang**

### CODE REVIEW

File: mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/apps/ForeignKeyClassifier.java [hunk 1]
```diff
@@ -129,33 +136,54 @@ protected void executeAppLogic() throws Exception {
...
+//        this.features.stream().forEach(feature -> feature.calcualteFeatureValue(dataset));
+
+        Dataset ds = new Dataset(dataset, features);
+        ds.buildDatasetStatistics();
+        ds.buildFeatureValueDistribution();
+
+        AbstractClassifier classifier = new NaiveBayes(ds);
+        classifier.train();
+
+//        // Set up the classifiers.
```
Comment (sekruse) [inline mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/apps/ForeignKeyClassifier.java]: Feel free to delete this code if it is no longer needed.

File: mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/apps/ForeignKeyClassifier.java [hunk 2]
```diff
@@ -379,7 +407,7 @@ public String explain(DependencyPrettyPrinter prettyPrinter) {
 
         @Parameter(names = {"--distinct-values"},
                 description = "ID of the constraint collection that contains the distinct value counts",
-                required = true)
+                required = false)
         public int dvcCollectionId;
```
Comment (sekruse) [inline mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/apps/ForeignKeyClassifier.java]: This parameter is probably not longer needed, is it? So we could also remove it...

File: mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/fk/Instance.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package de.hpi.isg.mdms.java.fk;
+
+import it.unimi.dsi.fastutil.objects.Object2ObjectArrayMap;
+
+import java.util.Map;
+
+/**
+ * Represent an instance in the dataset.
```
Comment (sekruse) [inline mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/fk/Instance.java]: Is each FK candidate an `Instance`?

Comment (lanchiang) [inline mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/fk/Instance.java]: Yes, i try to encapsulate each IND in the dataset as an Instance. So each FK candidate should also be an Instance

File: mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/apps/ForeignKeyClassifier.java (hunk 2 above)
Comment (lanchiang) [inline mdms/mdms-java/src/main/java/de/hpi/isg/mdms/java/apps/ForeignKeyClassifier.java]: Yes, I think so too:)

**Pull request merged**